libmscore/chord.cpp

```diff
@@ -3167,24 +3167,81 @@ TremoloChordType Chord::tremoloChordType() const
       return TremoloChordType::TremoloSingle;
       }
 
-
 //---------------------------------------------------------
 //   nextElement
 //---------------------------------------------------------
 
 Element* Chord::nextElement()
       {
-      for (int v = track() + 1; staffIdx() == v/VOICES; ++v) {
-            Element* e = segment()->element(v);
-            if (e) {
-                  if (e->type() == ElementType::CHORD)
-                        return static_cast<Chord*>(e)->notes().back();
-
-                  return e;
+      Element* e = score()->selection().element();
+      if (!e && !score()->selection().elements().isEmpty() )
+            e = score()->selection().elements().first();   
+      switch(e->type()) {
+            case ElementType::SYMBOL:
+            case ElementType::IMAGE:
+            case ElementType::FINGERING:
+            case ElementType::TEXT:
+            case ElementType::BEND: {
+                  Note* n = static_cast<Note*>(e->parent());
+                  if(n == _notes.front()) {                       
+                        if (_arpeggio)
+                              return _arpeggio;
+                        else if (_tremolo)
+                              return _tremolo;
+                        break;
+                        }
+                  auto i = std::find(_notes.begin(), _notes.end(), n);
```

Comment:
Not necessarily a problem but - I'm old enough that I still never really "got" iterators.  We use them occasionally in MsueScore but not much.  If this is the best way to get the job done, great, but if there another way, it might feel more consistent.